WordPress/gutenberg
# [RNMobile] Remove non-functioning caption button from mobile block toolbar

## What?
Removes a non-functioning caption button from the mobile Gallery block toolbar. On mobile, the caption can be edited directly via the text input beneath the gallery item, and the caption button is not utilized in the same way it is on web.

## Why?
A button that is used to add/remove gallery captions on the web is now present in the mobile gallery toolbar after recent [Editor UX changes](https://github.com/WordPress/gutenberg/pull/52289). This is very similar to the [MediaReplaceFlow component being removed](https://github.com/WordPress/gutenberg/pull/52966) from the toolbar for mobile.

This change removes (or hides) a non-functioning button on mobile as further discussion on the button's behavior continues: https://github.com/WordPress/gutenberg/issues/52977#issuecomment-1651813893

## How?
Moves the web's caption button into a `Platform.isWeb` boolean block. 

## Testing Instructions
1. Create a new post.
2. Add a gallery block.
3. Observe that there is no longer a non-functional button to add a caption, as in the left-hand screenshot below.

### Testing Instructions for Keyboard


## Screenshots or screencast 
Before | After
-|-
<img src="https://github.com/WordPress/gutenberg/assets/643285/8f40fad5-bd4d-44cf-8810-6ae0a021d5bb" width="350" /> | <img src="https://github.com/WordPress/gutenberg/assets/643285/2674384b-a9dc-4af4-8758-b88ace2825f6" width="350" />
